corehq/apps/app_manager/views/modules.py

```diff
@@ -777,26 +777,45 @@ def upgrade_shadow_module(request, domain, app_id, module_unique_id):
 @require_can_edit_apps
 def overwrite_module_case_list(request, domain, app_id, module_unique_id):
     app = get_app(domain, app_id)
-    source_module_unique_id = request.POST['source_module_unique_id']
-    source_module = app.get_module_by_unique_id(source_module_unique_id)
-    dest_module = app.get_module_by_unique_id(module_unique_id)
+    dest_module_unique_ids = request.POST.getlist('dest_module_unique_ids')
+    display_properties = request.POST.get('display_properties')
+    case_list_filter = request.POST.get('case_list_filter')
+    other_configuration = request.POST.get('other_configuration')
+    src_module = app.get_module_by_unique_id(module_unique_id)
     detail_type = request.POST['detail_type']
-    assert detail_type in ['short', 'long']
-    if not hasattr(source_module, 'case_details'):
-        messages.error(
-            request,
-            _("Sorry, couldn't find case list configuration for module {}. "
-              "Please report an issue if you believe this is a mistake.").format(source_module.default_name()))
-    elif source_module.case_type != dest_module.case_type:
-        messages.error(
-            request,
-            _("Please choose a module with the same case type as the current one ({}).").format(
-                dest_module.case_type)
-        )
-    else:
-        setattr(dest_module.case_details, detail_type, getattr(source_module.case_details, detail_type))
-        app.save()
-        messages.success(request, _('Case list updated form module {}.').format(source_module.default_name()))
+    for dest_module_unique_id in dest_module_unique_ids:
+        dest_module = app.get_module_by_unique_id(dest_module_unique_id)
+        assert detail_type in ['short', 'long']
+        if not hasattr(dest_module, 'case_details'):
+            messages.error(
+                request,
+                _("Sorry, couldn't find case list configuration for module {}. "
+                "Please report an issue if you believe this is a mistake.").format(dest_module.default_name()))
+        elif dest_module.case_type != src_module.case_type:
+            messages.error(
+                request,
+                _("Please choose a module with the same case type as the current one ({}).").format(
+                    src_module.case_type)
+            )
+        else:
+            if detail_type == 'long':
+                setattr(dest_module.case_details, detail_type, getattr(src_module.case_details, detail_type))
+            else:
```

Comment:
This else block is doing a lot. Can you split it into smaller, well-named functions? The other_configuration block in particular would be good to be pulled out as a helper function, possibly moved to the detail class itself in `app_manager/models.py`.

Reply:
Another reason to split this up is it would allow you to write tests for this functionality.

Reply:
Split into functions, let me know if it's ok or we can split more.